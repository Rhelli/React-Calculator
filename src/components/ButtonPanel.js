import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = props => {
  const handleClick = buttonName => {
    props.clickHandler(buttonName);
  };

  const group1 = ['AC', '+/-', '%', 'รท'];
  const group2 = ['7', '8', '9', 'x'];
  const group3 = ['4', '5', '6', '-'];
  const group4 = ['1', '2', '3', '+'];
  const group5 = ['0', '.', '='];

  return (
    <div>
      <div id="group1">
        { group1.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />) }
      </div>
      <div id="group2">
        {group2.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
      <div id="group3">
        {group3.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
      <div id="group4">
        {group4.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
      <div id="group5">
        {group5.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
    </div>
  );
};

ButtonPanel.defaultProps = {
  clickHandler: null,
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

export default ButtonPanel;
