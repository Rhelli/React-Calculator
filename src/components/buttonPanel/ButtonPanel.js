import React from 'react';
import PropTypes from 'prop-types';
import Button from '../button/Button';
import styles from './buttonPanel.module.scss';

const ButtonPanel = props => {
  const handleClick = buttonName => {
    props.clickHandler(buttonName);
  };

  const group1 = ['AC', '+/-', '%', 'รท'];
  const group2 = ['7', '8', '9', 'x'];
  const group3 = ['4', '5', '6', '-'];
  const group4 = ['1', '2', '3', '+'];
  const group5 = ['0', '.', '='];


  return (
    <div className={styles.buttonPanel}>
      <div id="group1" className={styles.row}>
        { group1.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />) }
      </div>
      <div id="group2" className={styles.row}>
        {group2.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
      <div id="group3" className={styles.row}>
        {group3.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
      <div id="group4" className={styles.row}>
        {group4.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
      <div id="group5" className={styles.row}>
        {group5.map(val => <Button clickHandler={handleClick} key={`btn_${val}`} buttonName={val} />)}
      </div>
    </div>
  );
};

ButtonPanel.defaultProps = {
  clickHandler: null,
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

export default ButtonPanel;
