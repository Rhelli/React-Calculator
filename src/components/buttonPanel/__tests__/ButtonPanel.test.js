import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ButtonPanel from '../ButtonPanel';

describe('The Button Panel Appearance', () => {
  it('Matches the Button Panel Snapshot', () => {
    const tree = renderer.create(<ButtonPanel />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders all of the buttons correctly', () => {
    const { getByText } = render(<ButtonPanel />);

    getByText('AC');
    getByText('+/-');
    getByText('%');
    getByText('รท');
    getByText('7');
    getByText('8');
    getByText('9');
    getByText('x');
    getByText('4');
    getByText('5');
    getByText('6');
    getByText('-');
    getByText('1');
    getByText('2');
    getByText('3');
    getByText('+');
    getByText('0');
    getByText('.');
    getByText('=');
  });
});
