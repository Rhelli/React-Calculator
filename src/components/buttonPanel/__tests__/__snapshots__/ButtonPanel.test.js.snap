// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button Panel Appearance Matches the Button Panel Snapshot 1`] = `
<div
  className="buttonPanel"
>
  <div
    className="row"
    id="group1"
  >
    <div
      className="button"
    >
      <button
        className="grey"
        id="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="orange"
        id="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
  </div>
  <div
    className="row"
    id="group2"
  >
    <div
      className="button"
    >
      <button
        className="grey"
        id="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="orange"
        id="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
  </div>
  <div
    className="row"
    id="group3"
  >
    <div
      className="button"
    >
      <button
        className="grey"
        id="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="orange"
        id="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
  </div>
  <div
    className="row"
    id="group4"
  >
    <div
      className="button"
    >
      <button
        className="grey"
        id="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="orange"
        id="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="row"
    id="group5"
  >
    <div
      className="zeroButton"
    >
      <button
        className="grey"
        id="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="grey"
        id="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="orange"
        id="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
